src/include/catalog/catalog.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+#include <memory>
+#include <unordered_map>
+#include "catalog/catalog_defs.h"
+#include "catalog/database_handle.h"
+#include "common/strong_typedef.h"
+#include "storage/sql_table.h"
+namespace terrier::catalog {
+
+class DatabaseHandle;
+/**
+ * The global catalog object. It contains all the information about global catalog tables. It's also
+ * the entry point for transactions to access any data in any sql table.
+ *
+ * OID assignment:
+ * Note that we do not have a concept of oid_t anymore. Instead, we have
+ *  db_oid_t, nsp_oid_t, table_oid_t, col_oid_t
+ * In addition, for nsp_oid_t, table_oid_t, and col_oid_t, we only guarantee uniqueness inside a database, which means
+ * that the table_oid for pg_attribute in database A could be the same as pg_attribute in database B.
+ *
+ * db_oid_t:
+ *  0          is reserved for default database - terrier
+ *  1+         are used for new databases
+ *
+ * nsp_oid_t:
```

Comment:
nsp is not an intuitive abbreviation. Expand all nsp abbreviations to namespace.

Reply:
Arguably, Postgres does it this way: https://www.postgresql.org/docs/11/catalog-pg-namespace.html

But I will expand it nevertheless.